api/base/serializers.py

```diff
@@ -328,7 +412,7 @@ def to_representation(self, obj, envelope='data'):
             except SkipField:
                 continue
 
-            if isinstance(field, JSONAPIHyperlinkedIdentityField):
+            if isinstance(field, HyperlinkedFieldMixin):
```

Comment:
Mixins are used for adding behavior, not for sharing types. Maybe check `JSONAPIHyperlinkedRelatedField` instead and make `JSONAPIHyperlinkedGuidRelatedField` a subclass of `JSONAPIHyperlinkedRelatedField`.